src/linker/Linker.Steps/BasePerAssemblyStep.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+
+	public abstract class BasePerAssemblyStep
+	{
+		protected readonly AssemblyDefinition _assembly;
+
+		protected LinkContext Context { get; private set; }
+
+		public AnnotationStore Annotations => Context?.Annotations;
+
+		public BasePerAssemblyStep (AssemblyDefinition assembly, LinkContext context)
```

Comment:
I wonder if we should take this a step further and simply remove the notion of "Steps" - rename - and remove the base class. There's no real need for it.

Even if keep the "Steps" notation, I don't like that we create it for each assembly (that's not how the new extensibility will work either). I would expect `Process(assembly)` style... or just make it static and get rid of "steps".

Reply:
I made them static classes instead, and got rid of the "Step" terminology.